README.md

```diff
@@ -1,3 +1,4 @@
+[![Open in Visual Studio Code](https://classroom.github.com/assets/open-in-vscode-2e0aaae1b6195c2367325f4f02e2d04e9abb55f0b24a779b69b11b9e10269abc.svg)](https://classroom.github.com/online_ide?assignment_repo_id=15535780&assignment_repo_type=AssignmentRepo)
 # SE_Day1
```

Comment:
Answer

Software Developer

Software developers, also referred to as coders, are your main and indispensable troops on a software development project of any complexity and scale. Team doesn’t roll without them. Their role on the project is pretty straightforward: writing and implementing clean, efficient code based on the provided technical requirements. It wouldn’t hurt to mention that the more effectively the management communicates the technical requirements to your dev team, the lower the risk of accumulating technical debt.

Again we can say Software developers are also known as product engineers, software developers use their programming skills to create software. They follow the guidelines established by the product owner.

They are also responsible for creating proper software documentation, and developing updates and upgrades, and resolving issues for existing software.

Quality Assurance

the role of Quality Assurance (QA) Software Engineers has become indispensable. As software applications continue to grow in complexity and importance across various industries, the need for reliable, secure, and high-quality software is paramount. QA Software Engineers play a crucial role in ensuring that software meets the desired standards of quality, functionality, and performance. This essay explores the responsibilities, skills, and challenges faced by QA Software Engineers in today's dynamic software development environment.

Responsibilities of QA Software Engineers Test Planning and Strategy: QA Software Engineers are responsible for developing comprehensive test plans and strategies that outline the testing process, scope, and objectives. These plans serve as roadmaps for the entire testing phase. Test Case Design and Execution: QA Engineers create test cases that cover various aspects of the software, including functional, usability, performance, and security testing. They execute these test cases systematically to identify defects and issues. Automation: In the modern software development landscape, automation has become a cornerstone of QA. QA Software Engineers use automation tools and frameworks to streamline repetitive testing tasks, increase test coverage, and improve efficiency. Bug Tracking and Reporting: QA Engineers identify, document, and report software defects to the development team. Effective communication and collaboration are key in this process to ensure that issues are resolved promptly. Continuous Integration and Continuous Delivery (CI/CD): QA Software Engineers often integrate testing into the CI/CD pipeline to ensure that code changes are thoroughly tested before deployment, promoting a culture of continuous improvement. Performance Testing: QA Engineers assess the software's performance under various conditions, helping to optimize its speed, scalability, and reliability. Security Testing: With the increasing threat of cyberattacks, QA Engineers play a critical role in identifying and mitigating security vulnerabilities within the software. What’s a project manager in software Engineering? Project managers in software provide consistent leadership and technical expertise to help their teams complete projects correctly and on time. During the software development process, project managers may help their teams design, execute, monitor and complete their work to ensure the clients receive high-quality software products. Software project managers typically lead the production of the deliverable software and the employees performing the project.
What does a project manager in software Engineering do? A software project manager's responsibilities often depend on the scope of the projects they're working on. Some of the job duties a software project manager may be responsible for include:

Communicating with clients Communication between a project manager and a client may be vital in ensuring the success of a project. The project manager can effectively communicate the project's objectives, progress and any problems that may cause delays in the project. This communication keeps the clients aware of the changes or potential issues on the project. Frequent communication can help build trust between the project manager and the client. This trust can help ensure that the manager guides the team in completing the project to the client's satisfaction.
Leading the team Software project managers assign tasks, evaluate their team's performance and provide guidance throughout the software development process. A project manager may possess strong leadership qualities, such as motivating and inspiring team members to a common goal. These qualities may help them effectively direct the team to achieve the client’s goals. Examples of ways the manager may lead the group include: Setting a clear vision and goal for the project Assigning tasks and responsibilities to team members Providing feedback and support to team members Resolving any conflict within the team that may arise during the project

Planning Project managers may use a variety of tools and methods to plan projects. They can first develop a clear understanding of the project objectives and then identify the steps to achieve the objectives. These professionals may also create schedules and budgets to help run the project. If issues arise during the project development, the managers can re-plan to allow solving of the risks and potential problems. A software project manager typically creates a detailed plan that outlines the tasks necessary to complete a software project. This plan can include the team's timelines, milestones and software development resources. Resources may include the hiring of new team members. The project manager then works with the team to execute the plan and track the progress of the software. This process allows software project managers to translate the client's requests into project plans. Related: Product Planning: Definition, Examples and Benefits

Serving as a liaison Software project managers serve as liaisons between the development team and the other stakeholders in a software project. They may be responsible for communicating project status, managing changes and requesting additional resources to help complete the project. Examples of stakeholders the project managers communicate with include the product owner, business analysts and the quality assurance team. This liaison allows clients and team members to discuss progress and make new project adjustments conveniently.

Testing and ensuring results Software project managers typically test products by developing plans based on the client's requirements. The managers can execute these plans at different phases of the project development. The test plans detail the steps necessary to verify the product's performance, analyze its functionality and identify potential issues. After executing the test plans, the project manager can review the results to ensure that the product meets the set requirements.